frontend/src/shared/utils/dates.ts

```diff
@@ -0,0 +1,14 @@
+// Function to convert date object to YYYY-MM-DD format
+export const formatDate = (date: Date) => {
+    const year = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(date);
+    const month = new Intl.DateTimeFormat('en', { month: 'numeric' }).format(date);
+    const day = new Intl.DateTimeFormat('en', { day: 'numeric' }).format(date);
```

Comment:
no change: but this might be simpler. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts